pandas/core/indexes/multi.py

```diff
@@ -698,17 +698,26 @@ def _values(self) -> np.ndarray:
         values = []
 
         for i in range(self.nlevels):
-            vals = self._get_level_values(i)
+            index = self.levels[i]
+            codes = self.codes[i]
+
+            vals = index
             if is_categorical_dtype(vals.dtype):
                 vals = cast("CategoricalIndex", vals)
                 vals = vals._data._internal_get_values()
+
+            if isinstance(vals, ABCDatetimeIndex):
```

Comment:
hmm this seems totally unecessary as there is an astype after

Reply:
The `.astype(object)` does not remove the freq on the underlying Timestamps. I had to add that to fix a failing to_json test due to the non-null freq. 

```
import pandas as pd

dates = pd.date_range('2000-01-01', periods=3)

mi = pd.MultiIndex.from_arrays(
    [dates, [1, 2, 3]]
)

print(dates.freq)
print(dates.astype(object)[0].freq)
print(mi._get_level_values(0).freq)
```

```
<Day>
<Day>
None
```